doc/code/event_system.md

```diff
@@ -0,0 +1,106 @@
+# Event System
+
+Game simulation in openage is event-driven. The internal event system
+manages the creation, scheduling, and execution of these events.
+
+1. [Architecture](#architecture)
+2. [Event Handler Setup](#event-handler-setup)
+3. [Event Instances](#event-instances)
+4. [Triggering/Targeting Events](#triggeringtargeting-events)
+5. [Canceling Events](#canceling-events)
+
+
+## Architecture
+
+![event system UML](images/event_classes_uml.svg)
+
+The central component of the event system is the *event loop* (`EventLoop`). It is
+the main point of interaction between the game simulation logic and the event system.
+New events (`Event` instances) are added via the `create_event(..)` method and scheduled based on the
+configuration of the associated *[event handler](#event-handler-setup)* (`EventHandler`).
+Internally, the event is stored inside an *event queue* (`EventQueue`) instance
+sorts the events by invoke time.
+
+Events are executed by passing the current [simulation time](time.md) to the
+event loop's `reach_time(..)` method. This executes all events with invoke time
+
+```
+t_old < t_e <= t_cur
+```
+
+where `t_cur` is the current simulation time, `t_old` is the last time `reach_time(..)` was
+called, and `t_e` is the event's scheduled invoke time. Events are executed in order of
+earliest to latest invoke time. During execution, events may trigger the rescheduling of
+other events if there are dependencies between the events.
+
+Both the creation and execution of events require passing a `State` object that functions
+as a persistant global storage for values that are not linked to a specific event executions.
+In practice, `State` is primarily used to pass the state of the [game simulation](/doc/code/game_simulation/)
+to the event system via an object of the derived `GameState` class. This object allows
+access to the indexing structures of the current game that can be used to retrieve
+the available game entities, for example.
+
+
+## Event Handler Setup
```

Comment:
the relation of events and event handlers and what the latter are used for is not clear for me

Reply:
rephrased it slightly with more explanation